ksp_plugin/pile_up.cpp

```diff
@@ -261,27 +266,56 @@ PileUp PileUp::ReadFromMessage(
 
   bool const is_pre_cartan = !message.has_adaptive_step_parameters() ||
                              !message.has_fixed_step_parameters();
+  bool const is_pre_cesàro = message.history().children().empty();
   std::unique_ptr<PileUp> pile_up;
-  if (is_pre_cartan) {
-    pile_up = std::unique_ptr<PileUp>(
-        new PileUp(std::move(parts),
-                   DefaultProlongationParameters(),
-                   DefaultHistoryParameters(),
-                   DiscreteTrajectory<Barycentric>::ReadFromMessage(
-                       message.psychohistory(),
-                       /*forks=*/{}),
-                   ephemeris));
+  if (is_pre_cesàro) {
+    if (is_pre_cartan) {
+      pile_up = std::unique_ptr<PileUp>(
+          new PileUp(std::move(parts),
+                     DefaultProlongationParameters(),
+                     DefaultHistoryParameters(),
+                     DiscreteTrajectory<Barycentric>::ReadFromMessage(
+                         message.history(),
+                         /*forks=*/{}),
+                     /*psychohistory=*/nullptr,
+                     ephemeris));
+    } else {
+      pile_up = std::unique_ptr<PileUp>(
+        new PileUp(
+            std::move(parts),
+            Ephemeris<Barycentric>::AdaptiveStepParameters::ReadFromMessage(
+                message.adaptive_step_parameters()),
+            Ephemeris<Barycentric>::FixedStepParameters::ReadFromMessage(
+                message.fixed_step_parameters()),
+            DiscreteTrajectory<Barycentric>::ReadFromMessage(
+                message.history(),
+                /*forks=*/{}),
+            /*psychohistory=*/nullptr,
+            ephemeris));
+    }
+    // Fork a psychohistory for compatibility if there is a non-authoritative
+    // point.
+    if (pile_up->history_->Size() == 2) {
+      Instant const history_begin_time = pile_up->history_->Begin().time();
+      pile_up->psychohistory_ =
+          pile_up->history_->NewForkWithCopy(history_begin_time);
+      pile_up->history_->ForgetAfter(history_begin_time);
+    }
```

Comment:
This means it is possible to have no psychohistory, if the history size was 1; on the other hand, AdvanceTime used to check that the history size was in [1,2], and now checks that the psychohistory is non-null. Always fork a psychohistory (I don't think we can easily make psychohistory_ not_null; if we can we should).

Add a test to exercise this (deserializing a pre-Cesàro message with a history of size 1, and calling AdvanceTime on it).